### ISSUE TITLE
clarification on script for computing visual complexity

### ISSUE THREAD

**Issue by bakachan19** [emlinking/visual-complexity#1]
Hi,
I am interested in using your work to compute the visual complexity of images, however, I am not sure which script I should be using.
Is  the algorithm in Section 4 of your report  implemented in the following script
```python meanshift-filtered.py [path-to-image-directory] [path-to-output-directory] [cdiff_thresh] [sdiff_thresh]```?
Moreover, in the script above, does ```num_distinct``` indicate the value of your visual complexity metric?

If not, could you point me to the script to use to compute the visual complexity of images?

Thank you.

**Comment by emlinking** [comment]
1. Is the algorithm in Section 4 of your report implemented in the following script
python meanshift-filtered.py [path-to-image-directory] [path-to-output-directory] [cdiff_thresh] [sdiff_thresh]?

Yes

2. Moreover, in the script above, does num_distinct indicate the value of your visual complexity metric?

Yes, but see [section 5.1](https://emlinking.github.io/files/dreu_report.pdf) for a possible normalization to apply to convert the metric into the (0,1) range.

**Comment by nikiguo93** [comment]
Hi, I have images from MS COCO datasets. I would like to evaluate the image complexity. Should I finetue the BERT? Or just used python meanshift-filtered.py [path-to-image-directory] [path-to-output-directory] [cdiff_thresh] [sdiff_thresh]? how to have [cdiff_thresh] and [sdiff_thresh] values?

**Comment by emlinking** [comment]
1. Evaluating image complexity doesn't require any access to fine-tuned models.
2. Yes, you can evaluate image complexity by running this script:
`python meanshift-filtered.py [path-to-image-directory] [path-to-output-directory] [cdiff_thresh] [sdiff_thresh]`
3. The [cdiff_thresh] and [sdiff_thresh] values are empirically determined. You can see the values we used in Section 5.1 (p. 7) here (sdiff_thresh = 0.05 and cdiff_thresh = 4): https://emlinking.github.io/files/dreu_report.pdf.

**Comment by nikiguo93** [comment]
Hi, after getting the complexity scores, how to check whether the scores are ok? Like, how to evaluate whether the cdiff_thresh and sdiff_thresh settings are ok

**Comment by nikiguo93** [comment]
I found that my complexity scores are all above 0.7.... and some even 1.0. Is that ok

**Comment by emlinking** [comment]
You can see page 7 of the report here for the expected distribution of complexity scores in MSCOCO, which is approximately normal in the range (0, 1): https://emlinking.github.io/files/dreu_report.pdf

Depending on what subset of images you are looking at from MSCOCO, high complexity scores might be reasonable.

Here's an example of an image with complexity score = 1: https://emlinking.github.io/images/bathroom.png

If you look at section 3 of the paper, you will see that we validated the cdiff/sdiff settings by using the SAVOIAS dataset.

**Comment by nikiguo93** [comment]
![train-5009_nsd-41311](https://github.com/user-attachments/assets/c4da1e4f-965b-465c-9de5-7a4ee2526c04)
Hi, is it ok if the score of this image is above 0.8?

**Comment by nikiguo93** [comment]
The num_dictinct = 1. and should the score be np.tanh(1) =0.76?

**Comment by nikiguo93** [comment]
oh, my mistakes

**Comment by emlinking** [comment]
![image](https://github.com/user-attachments/assets/14ec92c5-be23-4222-8b96-78db2a928de1)
See page 7 for how to normalize raw # distinct regions > score: https://emlinking.github.io/files/dreu_report.pdf
